src/Umbraco.Web.UI.Client/src/common/resources/templatequery.resource.js

```diff
@@ -0,0 +1,151 @@
+         * @methodOf umbraco.resources.templateQueryResource
+         * @function
+         *
+         * @description
+         * Called to get allowed properties
+         * ##usage
+         * <pre>
+         * templateQueryResource.getAllowedProperties()
+         *    .then(function(response) {
+         *
+         *    });
+         * </pre>
+         */
+        function getAllowedProperties() {
+            return umbRequestHelper.resourcePromise(
+                $http.get(
+                    umbRequestHelper.getApiUrl(
+                        "templateQueryApiBaseUrl",
+                        "GetAllowedProperties")),
+                'Failed to retrieve properties');
+        }
+
+        /**
+         * @ngdoc function
+         * @name umbraco.resources.templateQueryResource#getContentTypes
+         * @methodOf umbraco.resources.templateQueryResource
+         * @function
+         *
+         * @description
+         * Called to get content types
+         * ##usage
+         * <pre>
+         * templateQueryResource.getContentTypes()
+         *    .then(function(response) {
+         *
+         *    });
+         * </pre>
+         */
+        function getContentTypes() {
+            return umbRequestHelper.resourcePromise(
+                $http.get(
+                    umbRequestHelper.getApiUrl(
+                        "templateQueryApiBaseUrl",
+                        "GetContentTypes")),
+                'Failed to retrieve content types');
+        }
+
+        /**
+         * @ngdoc function
+         * @name umbraco.resources.templateQueryResource#getFilterConditions
+         * @methodOf umbraco.resources.templateQueryResource
+         * @function
+         *
+         * @description
+         * Called to the filter conditions
+         * ##usage
+         * <pre>
+         * templateQueryResource.getFilterConditions()
+         *    .then(function(response) {
+         *
+         *    });
+         * </pre>
+         */
+        function getFilterConditions() {
+            return umbRequestHelper.resourcePromise(
+                $http.get(
+                    umbRequestHelper.getApiUrl(
+                        "templateQueryApiBaseUrl",
+                        "GetFilterConditions")),
+                'Failed to retrieve filter conditions');
+        }
+
+        /**
+         * @ngdoc function
+         * @name umbraco.resources.templateQueryResource#postTemplateQuery
+         * @methodOf umbraco.resources.templateQueryResource
+         * @function
+         *
+         * @description
+         * Called to get content types
+         * ##usage
+         * <pre>
+         * var query = {
+         *     contentType: {
+         *         name: "Everything"
+         *      },
+         *      source: {
+         *          name: "My website"
+         *      },
+         *      filters: [
+         *          {
+         *              property: undefined,
```

Comment:
Do we actually send undefinied/null values to the endpoint as a valid object?
If not can we show a valid JSON object we should send to do a query